diff --git a/src/third_party/websocketpp/websocketpp/transport/asio/endpoint.hpp b/src/third_party/websocketpp/websocketpp/transport/asio/endpoint.hpp
index 8c8c2f2..f21d8c7 100644
--- a/src/third_party/websocketpp/websocketpp/transport/asio/endpoint.hpp
+++ b/src/third_party/websocketpp/websocketpp/transport/asio/endpoint.hpp
@@ -485,17 +485,25 @@ public:
     void listen(std::string const & host, std::string const & service,
         lib::error_code & ec)
     {
         using boost::asio::ip::tcp;
         tcp::resolver r(*m_io_service);
-        tcp::resolver::results_type endpoint_iterator = r.resolve(host, service);
-        tcp::resolver::results_type end;
-        if (endpoint_iterator == end) {
+        boost::system::error_code bec;
+        tcp::resolver::results_type results = r.resolve(host, service, bec);
+
+        if (bec) {
             m_elog->write(log::elevel::library,
-                "asio::listen could not resolve the supplied host or service");
+                "asio::listen could not resolve the supplied host or service: " + bec.message());
             ec = make_error_code(error::invalid_host_service);
             return;
         }
-        listen(*endpoint_iterator,ec);
+
+        if (results.empty()) {
+            m_elog->write(log::elevel::library,
+                "asio::listen resolved host but returned no endpoints");
+            ec = make_error_code(error::invalid_host_service);
+            return;
+        }
+        listen(results.begin()->endpoint(), ec);
     }
 
     /// Set up endpoint for listening on a host and service
@@ -919,12 +927,11 @@ public:
         if (m_alog->static_test(log::alevel::devel)) {
             std::stringstream s;
             s << "Async DNS resolve successful. Results: ";
 
-            boost::asio::ip::tcp::resolver::results_type it, end;
-            for (it = iterator; it != end; ++it) {
-                s << (*it).endpoint() << " ";
-            }
+            for (auto const& result : iterator) {
+                s << result.endpoint() << " ";
+            }
 
             m_alog->write(log::alevel::devel,s.str());
         }

